docs/gui/screens.md

```diff
@@ -4,73 +4,241 @@ Screens are typically the base of all Graphical User Interfaces (GUIs) in Minecr
+            10, 10
+        );
+
+        // Transform the rectangle to its appropriate location using the pose
+        rectangle = rectangle.transformMaxBounds(this.pose);
+
+        // If there is a scissor area defined, return the intersection of the two rectangles
+        // Otherwise, return the full bounds
+        return this.scissorArea != null
+            ? this.scissorArea.intersection(rectangle)
+            : rectangle;
+    }
+
+    @Override
+    @Nullable
+    public ScreenRectangle scissorArea() {
+        return this.scissorArea;
+    }
+
+    @Override
+    public RenderPipeline pipeline() {
+        return RenderPipelines.GUI;
+    }
+
+    @Override
+    public TextureSetup textureSetup() {
+        // Should generally specify at least one texture in Sampler0
+        return TextureSetup.noTexture();
+    }
+
+    @Override
+    public void buildVertices(VertexConsumer consumer, float z) {
+        // Build the vertices using the vertex format specified by the pipeline
+        // For GUI, uses quads with position and color
+        // Color must be in ARGB format
+        consumer.addVertexWith2DPose(this.pose, 0,  0,  z).setUv(0, 0).setColor(0xFFFFFFFF);
+        consumer.addVertexWith2DPose(this.pose, 0,  10, z).setUv(0, 1).setColor(0xFFFFFFFF);
+        consumer.addVertexWith2DPose(this.pose, 10, 10, z).setUv(1, 1).setColor(0xFFFFFFFF);
+        consumer.addVertexWith2DPose(this.pose, 10, 0,  z).setUv(1, 0).setColor(0xFFFFFFFF);
+    }
+});
+```
+
+### Element Ordering
+
+So far, the elements shown above have only been operating on XY coordinates. The Z coordinate, on the other hand, is automatically computed based upon the elements render order. Starting at a Z of `0`, each element is render `0.01` ahead of the previous. 3D elements are drawn to a 2D texture before being rendered to the screen, so Z-fighting rarely, if ever, occurs.
+
+During the render phase, each stratum is rendered in order, with the nodes in the node list rendered from 'down' to 'up'. But what about within a given node? This is handled via the `GuiRenderer#ELEMENT_SORT_COMPARATOR`, which sorts elements based on their `GuiElementRenderState#scissorArea`, `pipeline`, then `textureSetup`.
+
+:::warning
+Glyphs rendered for text are not sorted and will always render after all elements in the current node.
+:::
+
+Elements with no specified `scissorArea` will always be rendered first, followed by the top Y, the bottom Y, the left X, and finally the right X. If the `scissorArea` for two elements match, the the sort key of the `pipeline` (via `RenderPipeline#getSortKey`) will be used. The sort key is based on the order that the `RenderPipeline`s are built, which, in vanilla is the classloading of static constants within `RenderPipelines`. If the sort keys match, then the `textureSetup` is used. Elements with no specified `textureSetup` are ordered first, followed by the sort key (via `TextureSetup#getSortKey`) of texture elements.
```

Comment:
```suggestion
Elements with no specified `scissorArea` will always be rendered first, followed by the top Y, the bottom Y, the left X, and finally the right X. If the `scissorArea` for two elements match, the sort key of the `pipeline` (via `RenderPipeline#getSortKey`) will be used. The sort key is based on the order that the `RenderPipeline`s are built in, which in vanilla is the classloading of static constants within `RenderPipelines`. If the sort keys match, then the `textureSetup` is used. Elements with no specified `textureSetup` are ordered first, followed by the sort key (via `TextureSetup#getSortKey`) of texture elements.
```
Also, what does "the top Y, the bottom Y, the left X, and finally the right X" mean?